src/main/java/org/jenkinsci/test/acceptance/guice/Cleaner.java

```diff
@@ -66,15 +68,18 @@ public String toString() {
         });
     }
 
-    public void performCleanUp() {
+    public List<Throwable> performCleanUp() {
         LOGGER.info("Performing cleanup tasks in order: " + tasks);
+        List<Throwable> errors = new ArrayList<>();
         for (Statement task : tasks) {
             try {
                 task.evaluate();
             } catch (Throwable t) {
                 LOGGER.log(Level.SEVERE, task + " failed", t);
```

Comment:
Normally it's bad practice to both log and throw, but in this case the method is also called from the world cleaner thread and connection handling thread, both of which have no way to propagate errors, so logging is still useful here.